modules/ximgproc/src/edgepreserving_filter.cpp

```diff
@@ -0,0 +1,237 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level
+// directory of this distribution and at http://opencv.org/license.html.
+//
+//                    Created by Simon Reich
+//
+
+#include "precomp.hpp"
+
+namespace cv
+{
+namespace ximgproc
+{
+using namespace std;
+
+void edgepreservingFilter(const InputArray _src, OutputArray _dst, int d,
+                          double threshold)
+{
+	Mat src = _src.getMat();
+
+	// [re]create the output array so that it has the proper size and type.
+	_dst.create(src.size(), src.type());
+	Mat dst = _dst.getMat();
+	src.copyTo(dst);
+
+	CV_Assert(src.type() == CV_8UC3);
```

Comment:
Move this check in the beginning of function. Perhaps via `_src.type()`

Reply:
Done.